src/main/java/org/wso2/lsp4intellij/client/DefaultLanguageClient.java

```diff
@@ -52,9 +74,12 @@ public class DefaultLanguageClient implements LanguageClient {
     @NotNull
     private final ClientContext context;
     protected boolean isModal = false;
+    private final String extension;
+    private final HashMap<String, Tuple2<String, String>> progressNotificationItems = new HashMap<>();
 
-    public DefaultLanguageClient(@NotNull ClientContext context) {
+    public DefaultLanguageClient(@NotNull ClientContext context, String extension) {
```

Comment:
We don't need to pass the extension as a separate contructor param cuz we can get the same via the `context`.